src/datasource.ts

```diff
@@ -131,14 +131,16 @@ export class GenericDatasource {
         return target.target !== 'select metric';
       })
       .map((target) => {
-        let data = target.data;
+        let data = isUndefined(target.data) ? null : target.data;
 
         if (typeof data === 'string' && data.trim() === '') {
           data = null;
         }
 
-        if (data) {
-          data = JSON.parse(data);
+        if (data !== null) {
+          data = JSON.parse(
+            this.templateSrv.replace(data, options.scopedVars, 'regex'),
```

Comment:
I'm wondering if its a good idea to pass in data as string- won't that potentially break the json formatting? Aren't we forced here to call this multiple times per value or even per key and value?

Reply:
I'll try `json`, should be good enough then, WDYT?

Reply:
I guess we have to document our expectations for the data input. I assume the template service does not know/care about the input format, only about the output format.

Assume this json input:

    {
        "someVarValue": "$templateVar"
    }

if `$templateVar` is an array that will break as it becomes `"[foo,bar]"`. 

Using `json` the input should look something like this:

    {
        "someVarValue": $templateVar
    }

... and we can feed the entire json string into the template service. At least that's what I'd expect.

So for the docs we should mention that we use `json` expansion of the template variables or make this configurable as part of the data source.

Am I making any sense here?

Reply:
Yes, lemme add docs for this.

Reply:
Check now. I force pushed and split the work into commits so it should be easy to see changes.